packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -449,10 +444,13 @@ class AnalyzeCommand extends FlutterCommand {
   Future<int> _analyzeWatch() async {
     List<String> directories;
 
-    if (isFlutterRepo) {
+    if (argResults['flutter-repo']) {
+      String root = path.absolute(ArtifactStore.flutterRoot);
+
       directories = <String>[];
-      directories.addAll(_gatherProjectPaths(path.absolute('examples')));
-      directories.addAll(_gatherProjectPaths(path.absolute('packages')));
+      directories.addAll(_gatherProjectPaths(path.join(root, 'examples')));
+      directories.addAll(_gatherProjectPaths(path.join(root, 'packages')));
+      directories.addAll(_gatherProjectPaths(path.join(root, 'dev')));
```

Comment:
this doesn't exactly match the logic in the single-shot case (e.g. that takes into account manual-tests and layers and so on). any chance we can move these to use more common code to determine the paths? Or is the analyzer server doing things very differently than regular analyzer?